yt_dlp/extractor/youtube.py

```diff
@@ -627,32 +633,78 @@ def _extract_context(self, ytcfg=None, default_client='web'):
         if self._SAPISID is None:
-            yt_cookies = self._get_cookies('https://www.youtube.com')
             # Sometimes SAPISID cookie isn't present but __Secure-3PAPISID is.
+            # YouTube also falls back to __Secure-3PAPISID if SAPISID is missing.
             # See: https://github.com/yt-dlp/yt-dlp/issues/393
+
             sapisid_cookie = dict_get(
-                yt_cookies, ('__Secure-3PAPISID', 'SAPISID'))
+                yt_cookies, ('SAPISID', '__Secure-3PAPISID'))
             if sapisid_cookie and sapisid_cookie.value:
                 self._SAPISID = sapisid_cookie.value
-                self.write_debug('Extracted SAPISID cookie')
-                # SAPISID cookie is required if not already present
-                if not yt_cookies.get('SAPISID'):
-                    self.write_debug('Copying __Secure-3PAPISID cookie to SAPISID cookie')
-                    self._set_cookie(
-                        '.youtube.com', 'SAPISID', self._SAPISID, secure=True, expire_time=time_now + 3600)
-            else:
-                self._SAPISID = False
-        if not self._SAPISID:
-            return None
-        # SAPISIDHASH algorithm from https://stackoverflow.com/a/32065323
-        sapisidhash = hashlib.sha1(
-            f'{time_now} {self._SAPISID} {origin}'.encode()).hexdigest()
-        return f'SAPISIDHASH {time_now}_{sapisidhash}'
+
+        if self._1PSAPISID is None:
+            _1papisid_cookie = yt_cookies.get('__Secure-1PAPISID')
+            if _1papisid_cookie and _1papisid_cookie.value:
+                self._1PSAPISID = _1papisid_cookie.value
+
+        if self._3PSAPISID is None:
+            _3papisid_cookie = yt_cookies.get('__Secure-3PAPISID')
+            if _3papisid_cookie and _3papisid_cookie.value:
+                self._3PSAPISID = _3papisid_cookie.value
+
+    def _generate_sid_authorization(self, origin='https://www.youtube.com', user_session_id=None):
+        """
+        Generate API Session ID Authorization for Innertube requests. Assumes all requests are secure (https).
+        @param origin: Origin URL
+        @param user_session_id: Optional User Session ID
+        @return: Authorization header value
+        """
+
+        authorizations = []
+        additional_parts = {}
+        if user_session_id:
+            additional_parts['u'] = user_session_id
+
+        self._load_sid_cookies()
+
+        if self._SAPISID:
+            authorizations.append(
+                self._make_sid_authorization('SAPISIDHASH', self._SAPISID, origin, additional_parts))
+
+        if self._1PSAPISID:
+            authorizations.append(
+                self._make_sid_authorization('SAPISID1PHASH', self._1PSAPISID, origin, additional_parts))
+
+        if self._3PSAPISID:
+            authorizations.append(
+                self._make_sid_authorization('SAPISID3PHASH', self._3PSAPISID, origin, additional_parts))
+
+        if not authorizations:
+            return
```

Comment:
it's nice to be explicit when the return value matters IMO
```suggestion
            return None
```